grin/src/Transformations/ExtendedSyntax/Optimising/CopyPropagation.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE LambdaCase, TupleSections, ViewPatterns #-}
+module Transformations.ExtendedSyntax.Optimising.CopyPropagation where
+
+import Control.Monad.State
```

Comment:
Please use Strict version if you don't rely on laziness heavily.

Reply:
Is it beneficial to use the strict version instead? Maybe to avoid space leaks?

Reply:
Yes. We don't want to drag around thunks.